File: php/features.php
<?php
declare(strict_types=1);

// Coffee SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CoffeeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CoffeeBaseFeature();
            case "test":
                return new CoffeeTestFeature();
            default:
                return new CoffeeBaseFeature();
        }
    }
}

File: py/features.py
# Coffee SDK feature factory

from feature.base_feature import CoffeeBaseFeature
from feature.test_feature import CoffeeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CoffeeBaseFeature(),
        "test": lambda: CoffeeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
